Remoted integration tests T1: centralized configuration push (#1115)
wazuh/wazuh-qa: docs/tests/integration/test_remoted/test_agent_communication/test_shared_configuration.md

# Test shared configuration push
## Overview

> Check that manager push shared configuration to agents as expected.

Reviewer: ```suggestion
Check if the manager pushes shared configuration to agents as expected.
```

## Objective

> Confirm that there are no problems when the manager is supposed to push shared configuration to agents.

Reviewer: ```suggestion
To confirm there are no problems when the manager is supposed to push shared configuration to agents.
```

By default, agents send a keep alive message to the manager every 10 seconds. For each one of these messages,
the Manager needs to check if the configuration for agent's groups have changed. If so, then the Wazuh Manager 

> needs to push the new effective configuration (`merge.mg` file) for the agent. One `merge.mg` must be pushed only once.

Reviewer: ```suggestion
must push the new effective configuration (`merged.mg` file) for the agent. It must be pushed only once.
```

> If the agent is added to a new group, a new `merge.mg` file must be generated for this agent and the configuration
> must be pushed as well.

Reviewer: ```suggestion
If the agent is added to a new group, a new `merged.mg` file must be generated for this agent and pushed as well.
```

## General info

To avoid race conditions problems, the keep_alive module module on agent_simulator is disabled and the keep_alive 

> messages are sent manually. This way we avoid the scenario where keep_alive thread sends two messages with the same

Reviewer: ```suggestion
messages are sent manually. This avoids the scenario where keep_alive thread sends two messages with the same
```

deprecated checksum while the process_message thread updates such checksum. This would cause an error because the
manager would send to the agent the shared configuration twice and it shouldn't.

|Tier | Number of tests | Time spent |
|:--:|:--:|:--:|
| 0 | 3 | 3m |

## Expected behavior
- Fail if remoted doesn't send the up file command.
- Fail if remoted doesn't push the shared configuration when required.
  Fail if remoted doesn't send the close file command.
  Fail if remoted push the same shared configuration twice.
  Fail if remoted doesn't push a new shared configuration after adding the agent to a new group.
- Fail if remoted doesn't return an error message when the agent is disconnected.
- Fail if remoted couldn't connect with an active agent.
## Testing

> - Test getconfig request

Author: I'm sorry I think that forgot to edit this part. ty.

> - Test getconfig request

Author: https://github.com/wazuh/wazuh-qa/pull/1115/commits/17101949d04bb961c482d766751afe464c6159ad
